### PR TITLE
Change more rct_object_entry to ObjectEntryDescriptor

### PR DESCRIPTION
**Pull request by IntelOrca**
Cherry-picked mostly from `new-save-format`

### CODE REVIEW

File: src/openrct2-ui/windows/ObjectLoadError.cpp [hunk 1]
```diff
@@ -199,7 +201,7 @@ class ObjectDownloader
         }
 
         auto& entry = _entries[_currentDownloadIndex];
-        auto name = String::Trim(std::string(entry.name, sizeof(entry.name)));
+        auto name = String::Trim(std::string(entry.GetName()));
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/ObjectLoadError.cpp:204]: This assumes that the object download server recognises json-style object names. We need to make sure this is the case before the new save format is merged.

Comment (IntelOrca) [reply src/openrct2-ui/windows/ObjectLoadError.cpp:204]: Yes, I plan for it to use a different API for JSON object names.

**Comment by Gymnasiast** [comment]
Why did you close this?

**Comment by IntelOrca** [comment]
It is too out of date now with new-save-format branch. I have decided it is not worth the effort to backport.

**Pull request closed without merging**